tests/ignite/metrics/nlp/test_bleu.py

```diff
@@ -167,6 +266,42 @@ def test_bleu_batch():
     assert ref_2 == reference_bleu_score
 
 
+def test_bleu_batch_micro():
+    bleu = Bleu(ngram=4, average="micro")
+
+    # Batch size 3
+    hypotheses = [corpus.cand_1, corpus.cand_2a, corpus.cand_2b]
+    refs = [corpus.references_1, corpus.references_2, corpus.references_2]
+    bleu.update((hypotheses, refs))
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        reference_bleu_score = corpus_bleu(refs, hypotheses)
+    assert bleu.compute() == reference_bleu_score
+
+    assert bleu._corpus_bleu(refs, hypotheses) == reference_bleu_score
+
+
+@pytest.mark.parametrize(
+    "candidate, references",
+    [
+        (corpus.cand_1, corpus.references_1),
+        (corpus.cand_2a, corpus.references_2),
+        (corpus.cand_2b, corpus.references_2),
+        (corpus.cand_1, corpus.references_1),
+    ],
+)
+def test_n_gram_counter(candidate, references):
+    bleu = Bleu(ngram=4)
+    hyp_length, ref_length = bleu._n_gram_counter([references], [candidate], Counter(), Counter())
+    assert hyp_length == len(candidate)
+
+    ref_lens = (len(reference) for reference in references)
+    closest_ref_len = min(ref_lens, key=lambda ref_len: (abs(ref_len - len(candidate)), ref_len))
+
+    assert ref_length == closest_ref_len
+
+
 def _test_distrib_integration(device):
```

Comment:
```suggestion
def _test_macro_distrib_integration(device):
```

Could we add `_test_micro_distrib_integration(device)` ?